packages/wp-now/src/wp-now.ts

```diff
@@ -0,0 +1,299 @@
+		this.php.chdir(documentRoot);
+		this.php.writeFile(
+			`${documentRoot}/index.php`,
+			`<?php echo 'Hello wp-now!';`
+		);
+	}
+
+	mountWordpress(fileName = 'latest') {
+		const { documentRoot } = this.options;
+		const root = path.join(WORDPRESS_ZIPS_PATH, fileName, 'wordpress');
+		this.php.mount(
+			{
+				root,
+			},
+			documentRoot
+		);
+		this.php.writeFile(
+			`${documentRoot}/wp-config.php`,
+			this.php.readFileAsText(`${documentRoot}/wp-config-sample.php`)
+		);
+		this.updateFile(
+			`${documentRoot}/wp-config.php`,
+			(contents) =>
+				`<?php
+          define('WP_HOME', "${this.options.absoluteUrl}");
+          define('WP_SITEURL', "${this.options.absoluteUrl}");
+          ?>${contents}`
+		);
+		this.php.mkdirTree(`${documentRoot}/wp-content/mu-plugins`);
+		this.php.writeFile(
+			`${documentRoot}/wp-content/mu-plugins/0-allow-wp-org.php`,
+			`<?php
+      // Needed because gethostbyname( 'wordpress.org' ) returns
+      // a private network IP address for some reason.
+      add_filter( 'allowed_redirect_hosts', function( $deprecated = '' ) {
+          return array(
+              'wordpress.org',
+              'api.wordpress.org',
+              'downloads.wordpress.org',
+          );
+      } );`
+		);
+	}
+
+	async runCode(code) {
+		const result = await this.php.run({
+			code,
+		});
+		console.log(result.text);
+		return result;
+	}
+
+	mountSqlite() {
+		const { documentRoot } = this.options;
+		const sqlitePluginPath = `${documentRoot}/wp-content/plugins/${SQLITE_FILENAME}`;
+		this.php.mkdirTree(sqlitePluginPath);
+		this.php.mount(SQLITE_PATH, sqlitePluginPath);
+		this.php.writeFile(
+			`${documentRoot}/wp-content/db.php`,
+			this.php
+				.readFileAsText(`${sqlitePluginPath}/db.copy`)
+				.replace(
+					/\{SQLITE_IMPLEMENTATION_FOLDER_PATH\}/g,
+					`${documentRoot}/wp-content/plugins/${SQLITE_FILENAME}`
+				)
+				.replace(/\{SQLITE_PLUGIN\}/g, SQLITE_FILENAME)
+		);
+	}
+
+	static #getWpContentHomePath(projectPath: string) {
+		const basename = path.basename(projectPath);
+		const directoryHash = crypto
+			.createHash('sha1')
+			.update(projectPath)
+			.digest('hex');
+		return path.join(
+			WP_NOW_PATH,
+			'wp-content',
+			`${basename}-${directoryHash}`
+		);
+	}
+
+	static #isPluginDirectory(projectPath: string): Boolean {
+		const files = fs.readdirSync(projectPath);
+		for (const file of files) {
+			if (file.endsWith('.php')) {
+				const fileContent = fs.readFileSync(
+					path.join(projectPath, file),
+					'utf8'
+				);
+				if (fileContent.includes('Plugin Name:')) {
+					return true;
+				}
+			}
+		}
+		return false;
+	}
+
+	static #isThemeDirectory(projectPath: string): Boolean {
+		const styleCSSExists = fs.existsSync(
+			path.join(projectPath, 'style.css')
+		);
+		if (!styleCSSExists) {
+			return false;
+		}
+		const styleCSS = fs.readFileSync(
+			path.join(projectPath, 'style.css'),
+			'utf-8'
+		);
+		return styleCSS.includes('Theme Name:');
```

Comment:
same note here as with the plugin search above. it's pretty heavy to _start_ by reading every file into memory and then searching for a basic string when we know:

 - the string should be close to the top
 - we only need as much data in memory as we need to find the string

Reply:
I agree. I'm using the same issue for theme and plugin mode optimization: https://github.com/WordPress/wordpress-playground/issues/227